apps/reaper/lib/reaper/xml/xml_stream/sax_handler.ex

```diff
@@ -0,0 +1,133 @@
+defmodule XMLStream.SaxHandler do
+  @moduledoc false
+
+  defmodule State do
+    @moduledoc false
+    @enforce_keys [:stack, :accumulate, :tag_stack, :tag_path, :emitter]
+    defstruct stack: [], accumulate: false, tag_stack: [], tag_path: [], emitter: nil
+
+    def new(fields \\ %{}) do
+      struct(__MODULE__, fields)
+    end
+
+    def update(state, fields) do
+      struct!(state, fields)
+    end
+  end
+
+  @behaviour Saxy.Handler
+  @file_chunk_size 40_000
+
+  ######################
+  ## Client Functions ##
+  ######################
+  def start_stream(path, selector, callback) do
+    selector = parse_selector(selector)
+
+    path
+    |> File.stream!([], @file_chunk_size)
+    |> handle_ufeff()
+    |> Saxy.parse_stream(
+      __MODULE__,
+      __MODULE__.State.new(tag_path: selector, emitter: callback)
+    )
+  end
+
+  ###############
+  ## Callbacks ##
+  ###############
+  def handle_event(:start_document, _, state) do
+    ok(state)
+  end
+
+  def handle_event(:end_document, _, state) do
+    ok(state)
+  end
+
+  def handle_event(:start_element, {tag_name, attributes}, %State{} = state) do
```

Comment:
Oops I missed that. They should all be `%State{}`. Mostly just for clarity and because it should make the handler fail fast in development if we mess up a return value.